### PR TITLE
FFI: Add +optAutoReleaseReturnedValue option

### PR DESCRIPTION
**Pull request by tinchodias**
Add +optAutoReleaseReturnedValue option.

This simplifies a common case where the C function returns an object what must be freed/released/destroyed by the caller.

Example of bindings to [a Cairo function](https://www.cairographics.org/manual/cairo-Paths.html#cairo-copy-path) as it's currently done in [Alexandrie](https://github.com/pharo-graphics/Alexandrie):

```
AeCairoContext>>
currentPath
	"Return a copy of the current path."

	^ self unownedCurrentPath
		  autoRelease;
		  yourself
```
and:
```
AeCairoContext>>
unownedCurrentPath
	"Return a copy of the current path.
	Warning: Sender is responsible of freeing (or sending autoRelease) the returned object."

	^ self ffiCall: #( AeCairoPath cairo_copy_path #( self ) )
```

With this new option, both methods can be rewritten as a single method:
```
AeCairoContext>>
currentPath
	"Return a copy of the current path."

	^ self
		  ffiCall: #( AeCairoPath cairo_copy_path #( self ) )
		  options: #( + optAutoReleaseReturnedValue )
```

### CODE REVIEW

File: src/UnifiedFFI/ExternalObject.extension.st [hunk 1]
```diff
@@ -2,8 +2,9 @@ Extension { #name : #ExternalObject }
 
 { #category : #'*UnifiedFFI' }
 ExternalObject >> autoRelease [
-	self isExternalAddress ifFalse: [ ^ self ].
-	^ self getHandle autoRelease
+	self getHandle isExternalAddress ifFalse: [ ^ self ].
+	self getHandle autoRelease.
+	^ self
```
Comment (tinchodias) [inline src/UnifiedFFI/ExternalObject.extension.st:7]: this fix is the only change that can affect users that don't explicitly set the new option

**Comment by tinchodias** [comment]
Programmed with @tesonep !

**Comment by MarcusDenker** [comment]
- [CI] test runs sometimes aborted due to SubscriptOutOfBounds error #12502

**Comment by jecisc** [comment]
Maybe this should be rebased for Pharo 12?

**Comment by tinchodias** [comment]
Hi, I don't know current status of Pharo 11, but I guess so.

**Comment by MarcusDenker** [comment]
I changed it to pharo12

**Pull request merged**